tests/transport/test_barrier_mt.cpp

```diff
@@ -0,0 +1,112 @@
+/* 
+ * GridTools
+ * 
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ * 
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * 
+ */
+#include <ghex/threads/std_thread/primitives.hpp>
+#include <ghex/threads/atomic/primitives.hpp>
+#include <ghex/threads/omp/primitives.hpp>
+#include <iostream>
+#include <iomanip>
+
+#include <gtest/gtest.h>
+
+#ifdef GHEX_TEST_USE_UCX
+#include <ghex/transport_layer/ucx/context.hpp>
+using transport = gridtools::ghex::tl::ucx_tag;
+#else
+#include <ghex/transport_layer/mpi/context.hpp>
+using transport = gridtools::ghex::tl::mpi_tag;
+#endif
+
+using threading = gridtools::ghex::threads::std_thread::primitives;
+using threading2 = gridtools::ghex::threads::atomic::primitives;
+using threading3 = gridtools::ghex::threads::omp::primitives;
+using context_type = gridtools::ghex::tl::context<transport, threading>;
+
+TEST(transport, barrier_mt_std) {
+    auto context_ptr = gridtools::ghex::tl::context_factory<transport,threading>::create(4, MPI_COMM_WORLD);
+    auto& context = *context_ptr;
+    
+    auto func = [&context]() {
+        auto token = context.get_token();
+        auto comm = context.get_communicator(token);
+
+        for(int i=0; i<50; i++)  {
```

Comment:
it is already quite slow. I'll try to restructure the test

Reply:
It is slow for the busy-wait based mutexes on my machine (since I am using more threads as I have physical cores - take into account, that mpi itself is spawning several processes itself). The only variant that is performant for me is `std::mutex` (the first test). I guess we have the following options
- just wait forever on laptops
- test only 50 iterations
- apply some cmake logic to configure the tests

Reply:
For now I guess it's ok, if there is a bug in a barrier we should be able to catch it sooner or later...